rpc/ethereum/namespaces/eth/api.go

```diff
@@ -721,7 +714,8 @@ func (e *PublicAPI) GetTransactionByBlockNumberAndIndex(blockNum rpctypes.BlockN
 	}
 
 	i := int(idx)
-	if i >= len(resBlock.Block.Txs) {
+	ethMsgs := e.getEthereumMsgFromTendermintBlock(resBlock)
+	if i >= len(ethMsgs) {
```

Comment:
L723 (`txBz := resBlock.Block.Txs[i]` ) is inconsistent with the L685. See comment about assuming the index corresponds to an ethereum tx

Reply:
line  723-730 needs to be removed @crypto-facs

Reply:
Good catch! If we agree this https://github.com/tharsis/ethermint/pull/741#discussion_r747546276 is the right approach to handle index across the endpoints I will add the filter here too.